src/main/java/wolox/training/repositories/IUserRepository.java

```diff
@@ -1,8 +1,15 @@
 package wolox.training.repositories;
 
+import java.util.Optional;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
-import wolox.training.models.Users;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+import wolox.training.models.User;
 
-public interface IUserRepository extends CrudRepository<Users, Long> {
+@Repository
+public interface IUserRepository extends CrudRepository<User, Long> {
 
+    @Query(value = "SELECT * FROM users WHERE user_username=:userUsername", nativeQuery = true)
+    Optional<User> findByUserUsername(@Param("userUsername") String username);
```

Comment:
The native query is not necessary. Check it out to [query creation documentation](https://docs.spring.io/spring-data/data-jpa/docs/current/reference/html/#jpa.query-methods.query-creation) to know about the keywords list you can use to build queries without using native queries.
```suggestion
    @Query(value = "SELECT * FROM users WHERE user_username=:userUsername", nativeQuery = true)
    Optional<User> findByUserUsername(String username);
```